cmd/s3-gw/app.go

```diff
@@ -107,10 +107,25 @@ func newApp(ctx context.Context, log *Logger, v *viper.Viper) *App {
 	}
 
 	neofsCfg := neofs.Config{
-		MaxObjectSize: int64(ni.MaxObjectSize()),
+		MaxObjectSize:        int64(ni.MaxObjectSize()),
+		IsSlicerEnabled:      v.GetBool(cfgSlicerEnabled),
+		IsHomomorphicEnabled: !ni.HomomorphicHashingDisabled(),
 	}
 
-	neoFS := neofs.NewNeoFS(conns, signer, anonSigner, neofsCfg)
+	// If slicer is disabled, we should use "static" getter, which doesn't make periodic requests to the NeoFS.
+	var epochGetter neofs.EpochGetter = ni
+
+	if neofsCfg.IsSlicerEnabled {
+		epochUpdateInterval := v.GetDuration(cfgEpochUpdateInterval)
+
+		if epochUpdateInterval == 0 {
+			epochUpdateInterval = defaultEpochUpdateInterval
```

Comment:
Likely we can calculate a half of blocks per epoch times time per block for the default, all of this data is available from the network. And it'd be the best one.

Reply:
Replaced